src/lib.rs

```diff
@@ -291,14 +309,14 @@ impl Manifest {
     /// Get the name that also refers to the input at the given location.
     pub fn get_input_name<'a>(&'a self, location: Location) -> Option<&'a str> {
         self.var_name_map.iter()
-            .find_map(|x| if let ResourceLocator::Input(loc) = x.1 {
+            .find_map(|x| if let ResourceLocator::Input(loc, _comp) = x.1 {
                 if *loc == location { Some(x.0.as_ref()) } else { None }
             } else { None })
     }
     /// Get the name that also refers to the output at the given location.
     pub fn get_output_name<'a>(&'a self, location: Location) -> Option<&'a str> {
         self.var_name_map.iter()
-            .find_map(|x| if let ResourceLocator::Output(loc) = x.1 {
+            .find_map(|x| if let ResourceLocator::Output(loc, _comp) = x.1 {
                 if *loc == location { Some(x.0.as_ref()) } else { None }
             } else { None })
     }
```

Comment:
No matter whether we have a united type for location-component representation, we should have access to every interface variable. We should be able to enumerate all input/output variables, and to get interface variable names with returned location-component information, so `component` has to be a parameter in these query methods somehow.